eudemo/eudemo/vacuum_vessel.py

```diff
@@ -107,9 +107,14 @@ def build_xz(
         """
         Build the x-z components of the vacuum vessel.
         """
-        inner_vv = offset_wire(
-            self.ivc_koz, self.params.g_vv_bb.value, join="arc", open_wire=False
+        inner_vv = _offset_wire_discretised(
+            self.ivc_koz,
+            self.params.g_vv_bb.value,
+            join="arc",
+            open_wire=False,
+            ndiscr=600,
         )
+
```

Comment:
I can't help but feeling that we're going to want to fallback onto this the vast majority of the time. It might be time to revive `MixedWireMaker`, because otherwise these massive polygons are going to be dreadfully slow and ugly.

Reply:
Also, why is the automatic fallback failing? Is the wire produced allegedly valid, or are we not catching an error or what?

Reply:
> I can't help but feeling that we're going to want to fallback onto this the vast majority of the time. It might be time to revive `MixedWireMaker`, because otherwise these massive polygons are going to be dreadfully slow and ugly.
It does feel like that could be a way out of this until #1347 is sorted

> Also, why is the automatic fallback failing? Is the wire produced allegedly valid, or are we not catching an error or what?

The wire is valid but it has a loop in it.

This is the surface that gets produced
![Screenshot vv](https://user-images.githubusercontent.com/81617086/221905220-1da19935-0747-4d5d-9560-048f363fecc5.png)

which is made from a wire that is first offset and then vary offset.
from left to right; variable offset, offset_wire, original wire
![Screenshot vv3](https://user-images.githubusercontent.com/81617086/221905210-ae0cd666-c63d-4011-87f2-6ab0aa435712.png)

Reply:
That's wild... we could try increasing the wire fix tolerance in this case though. But anyway, this will do for now.